### DISCUSSION TITLE
for long facet many sub-frames graph, wonder is there an option to enable showing x scale on every frame?

### DISCUSSION THREAD

**Discussion by tx0c** [observablehq/plot#623] [Q&A]
like this graph, I'd like to make xScale tick showing on every frame, because when the graph is long, more than 10 or so, it's very hard to read xScale, have to scroll back and forth, not convenient

And, when the yScale have different extent for each frame, is there an easy option to turn on different yScale for each frame? like in this graph, the 4th `transaction` has much lower values than others, using the same [0,4000] does not make sense

![image](https://user-images.githubusercontent.com/49659410/146656438-f20522e3-fde1-44af-9f6a-203d8bbbf957.png)

**Comment by Fil** [comment]
No, that's not possible with the current faceting system; in this case I'd suggest to generate multiple charts.

**Reply by tx0c** [reply]
then can convert this to 2 Feature Request ?

1. Plot.frame() currently only take `stroke` parameter to customize, but can consider more options like dup ticks?
2. facet can take more options, to infer value ranges from each facet rather than the whole data set?

**Reply by mbostock** [reply]
There is already a feature request for facets with varying scales. If you are interested in it, please upvote. Thanks.

https://github.com/observablehq/plot/issues/8

**Reply by tx0c** [reply]
but the other one is not there yet? this can be both x or y
1. when facet y has too many values, the graph becomes too tall, then the bottom frame might be out of screen, it's not convenient to scroll back and forth just to check which tick is what value;
2. same applies when facet x has too many values, the graph becomes too wide, ... would have to scroll the graph left and right ...

there should be a feature request to duplicate all tick labels showing on every frame? and this shouldn't be too hard to implement?

**Reply by Fil** [reply]
I think it's worth exploring the possibilities for axis duplication, but before implementing anything the difficult part is going to design the feature.

For example, one might want to repeat the x axis every 5 facets—this means that we might have to create a new type of option for the fy scale (band scales), in order to reserve a bit of space for the axis every 5 ticks.

A workaround is possible, though, using custom marks: https://observablehq.com/@fil/plot-multiple-axes

**Reply by tx0c** [reply]
> repeat the x axis every 5 facets

Or an option to repeat for every `500px` given number of px dimension? to describe only dup when more than a screen? e.g. for targeting audience with medium screen sizes (1024x768)

**Comment by mbostock** [comment]
This is possible using the **facetAnchor** option, such as `Plot.axisX({facetAnchor: null})` to draw the x-axis on every facet. https://observablehq.com/plot/marks/axis
